prow/cmd/webhook-server/mutation.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+	"gomodules.xyz/jsonpatch/v2"
+	"k8s.io/api/admission/v1beta1"
+	apiv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/test-infra/prow/apis/prowjobs/v1"
+)
+
+func (wa *webhookAgent) serveMutate(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		logrus.WithError(err).Fatal("Unable to read request")
+	}
+	admissionReview := &v1beta1.AdmissionReview{}
+	err = json.Unmarshal(body, admissionReview)
+	if err != nil {
+		logrus.WithError(err).Fatal("Unable to unmarshal admission review request")
+	}
+	admissionRequest := admissionReview.Request
+	var prowJob v1.ProwJob
+	err = json.Unmarshal(admissionRequest.Object.Raw, &prowJob)
+	if err != nil {
+		logrus.WithError(err).Fatal("Unable to unmarshal admission review request")
+	}
+	var admissionResponse *v1beta1.AdmissionResponse
+	var mutatedProwJobPatch []byte
+	logrus.Info("%v", wa.def.UtilityImages.CloneRefs)
+	if admissionRequest.Operation == "CREATE" {
+		mutatedProwJobPatch, err = mutateProwJob(&prowJob, wa.def)
+		if err != nil {
+			logrus.WithError(err).Fatal("Unable to return mutated prowjob")
+		}
+	}
+	admissionResponse = createMutatingAdmissionResponse(admissionRequest.UID, mutatedProwJobPatch)
```

Comment:
no need for this var. Just `admissionReview.Response = createMutatinAdmissionResponse(...)`